industrial_extrinsic_cal/src/calibration_job_definition.cpp

```diff
@@ -32,10 +32,81 @@ using std::string;
 using boost::shared_ptr;
 using boost::make_shared;
 using ceres::CostFunction;
+using industrial_extrinsic_cal::covariance_requests::CovarianceRequestType;
 
 namespace industrial_extrinsic_cal
 {
-
+  
+  /** @brief, a function used for debugging, and generating output for post processing */
+#define WRITE_DEBUG true
+#define OUTPUT_DEBUG_FILE "/home/nist/post_processing_data.txt"
+  void writeObservationData(std::string file_name, std::vector<ObservationDataPointList> odl)
+{
+  FILE *fp=NULL;
+  fp = fopen(file_name.c_str(), "w");
+  if(fp == NULL){
+    ROS_ERROR("Could not open %s", file_name.c_str());
+    return;
+  }
+  
+  int scene_id=-1;  
+  ROS_ERROR("processing %d lists", (int) odl.size());
+  for(int i=0; (int) i<odl.size(); i++){
+    if(odl[i].items_.size()>0){
+      scene_id = odl[i].items_[0].scene_id_;
+      fprintf(fp, "scene_id = %d \n", scene_id);
+      Pose6d t_pose;
+      t_pose.ax = odl[i].items_[0].target_pose_[0];
+      t_pose.ay = odl[i].items_[0].target_pose_[1];
+      t_pose.az = odl[i].items_[0].target_pose_[2];
+      t_pose.x  = odl[i].items_[0].target_pose_[3];
+      t_pose.y  = odl[i].items_[0].target_pose_[4];
+      t_pose.z  = odl[i].items_[0].target_pose_[5];
+      tf::Matrix3x3 basis = t_pose.getBasis();
+      fprintf(fp, "target_pose = [ %f %f %f %f;\n %f %f %f %f;\n %f %f %f %f;\n %f %f %f %f];\n",
+	      basis[0][0],basis[0][1], basis[0][2],t_pose.x,
+	      basis[1][0],basis[1][1], basis[1][2],t_pose.y,
+	      basis[2][0],basis[2][1], basis[2][2],t_pose.z,
+	      0.0, 0.0, 0.0, 1.0);
+      fprintf(fp, "cameras = [ ");
+      std::string camera_name("NULL");
+      for(int j=0; (int) j<odl[i].items_.size(); j++){
+	if(camera_name !=  odl[i].items_[j].camera_name_){
+	  fprintf(fp,"%s ", odl[i].items_[j].camera_name_.c_str());
+	}
+	camera_name =  odl[i].items_[j].camera_name_;
+      }
+      fprintf(fp, "]\n");
+    }
+  }
+  fclose(fp);
+}
+  CovarianceRequestType intToCovRequest(int request)
```

Comment:
This method is extremely fragile.  Changes to the header where this is defined will break this method (although nobody will know).